server/server.go

```diff
@@ -364,17 +398,26 @@ func (server *Server) checkNetIsolation() {
 
 // handleSessionClose handles the closing of a session. It removes the player of the session from the server.
 func (server *Server) handleSessionClose(controllable session.Controllable) {
+	if p, ok := server.p[controllable.UUID()]; ok {
```

Comment:
This map access does need to be protected by the Mutex lock. If you just change it to get p, ok := server.p[...] inside of the mutex lock and save the data outside of it, you should be good to go.

Reply:
woops ur right 😅